Userland/Libraries/LibGfx/ImageFormats/JPEGWriter.cpp

```diff
@@ -138,18 +138,32 @@ class JPEGEncodingContext {
 
     void fdct_and_quantization()
     {
+        constexpr double pi_over_16 = AK::Pi<double> / 16;
+        static Optional<Array<double, 106>> maybe_cos_lut = OptionalNone {};
```

Comment:
I can't comment on the math, but instead of awkwardly storing this in an Optional and checking it every time, we do this everywhere else that we create a static table:

```c++
static Array<double, 106> create_cosine_lookup_table()
{
    static constexpr double pi_over_16 = AK::Pi<double> / 16;
    
    Array<double, 106> table;
    table.fill(NAN);

    for (u8 u = 0; u < 8; ++u) {
        for (u8 x = 0; x < 8; ++x)
            table[(2 * x + 1) * u] = cos((2 * x + 1) * u * pi_over_16);
    }

    return table;
}

void fdct_and_quantization()
{
    static auto cosine_table = create_cosine_lookup_table();

}
```

Reply:
Looks so obvious now, I definitely did forget how static initialization work :sweat_smile: